src/Approximations/template_directions.jl

```diff
@@ -374,8 +407,80 @@ SphericalDirections(Nθ::Int, Nφ::Int) = SphericalDirections{Float64}(Nθ, Nφ)
 Base.eltype(::Type{SphericalDirections{N}}) where {N} = Vector{N}
 Base.length(sd::SphericalDirections) = length(sd.stack)
 dim(::SphericalDirections) = 3
+isbounding(sd::SphericalDirections) = sd.Nθ > 2 && sd.Nφ > 2
 
 function Base.iterate(sd::SphericalDirections, state::Int=1)
     state == length(sd.stack)+1 && return nothing
     return (sd.stack[state], state + 1)
 end
+
+# ==================================================
+# Custom template directions
+# ==================================================
+
+"""
+    CustomDirections{N} <: AbstractDirections{N}
+
+User-defined template directions.
+
+### Fields
+
+- `directions` -- list of template directions
+- `n`          -- dimension
+- `isbounding` -- boundedness status
+
+### Notes
+
+Since custom directions may not be bounded, boundedness can either be asserted
+in the constructor or it will be determined automatically by performing an
+experiment (more precisely, we overapproximate the unit ball in the infinity
+norm using the given directions).
+
+The dimension will also be determined automatically, unless the empty vector is
+passed (in which case the optional argument `n` needs to be specified).
+"""
+struct CustomDirections{N} <: AbstractDirections{N}
+    directions::AbstractVector{<:AbstractVector{N}}
+    n::Int
+    isbounding::Bool
+
+    CustomDirections{N}(directions::AbstractVector{<:AbstractVector{N}};
+                        n::Int=determine_dimension(directions),
+                        isbounding::Bool=determine_boundedness(directions)
+                       ) where {N} = new{N}(directions, n, isbounding)
+end
+
+# convenience constructor
+CustomDirections(directions::AbstractVector{<:AbstractVector{N}};
+                 n::Int=determine_dimension(directions),
+                 isbounding::Bool=determine_boundedness(directions)) where {N} =
+    CustomDirections{N}(directions; n=n, isbounding=isbounding)
+
+function determine_dimension(directions)
+    isempty(directions) && throw(ArgumentError("empty template directions " *
+                                               "need a specified dimension"))
+    return length(directions[1])
+end
+
+function determine_boundedness(directions::AbstractVector{<:AbstractVector{N}}
+                              ) where {N}
+    isempty(directions) && return false
+
+    # determine boundedness from a trial with a BallInf
+    set = BallInf(zeros(N, length(directions[1])), N(1))
```

Comment:
What if we check boundedness of the polyhedron whose defining halfspaces have normal directions `directions` and the displacement from the origin is 1:

```julia
n = length(first(directions))
isbounded(HPolyhedron(vcat(directions'...), ones(n)))
```